api/api_integration_test.go

```diff
@@ -0,0 +1,117 @@
+package api_test
+
+import (
+	"database/sql"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/builtin/logical/pki"
+	"github.com/hashicorp/vault/builtin/logical/transit"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+
+	cleanhttp "github.com/hashicorp/go-cleanhttp"
+	vaulthttp "github.com/hashicorp/vault/http"
+	logxi "github.com/mgutz/logxi/v1"
+	dockertest "gopkg.in/ory-am/dockertest.v3"
+)
+
+var testVaultServerDefaultBackends = map[string]logical.Factory{
+	"transit": transit.Factory,
+	"pki":     pki.Factory,
+}
+
+func testVaultServer(t testing.TB) (*api.Client, func()) {
+	return testVaultServerBackends(t, testVaultServerDefaultBackends)
+}
+
+func testVaultServerBackends(t testing.TB, backends map[string]logical.Factory) (*api.Client, func()) {
+	handlers := []http.Handler{
+		http.NewServeMux(),
+		http.NewServeMux(),
+		http.NewServeMux(),
+	}
+
+	coreConfig := &vault.CoreConfig{
+		DisableMlock:    true,
+		DisableCache:    true,
+		Logger:          logxi.NullLog,
+		LogicalBackends: backends,
+	}
+
+	// Chicken-and-egg: Handler needs a core. So we create handlers first, then
+	// add routes chained to a Handler-created handler.
+	cores := vault.TestCluster(t, handlers, coreConfig, true)
+	for i, core := range cores {
+		handlers[i].(*http.ServeMux).Handle("/", vaulthttp.Handler(core.Core))
+	}
+
+	// make it easy to get access to the active
+	core := cores[0].Core
+	vault.TestWaitActive(t, core)
+
+	rootToken := cores[0].Root
+	address := fmt.Sprintf("https://127.0.0.1:%d", cores[1].Listeners[0].Address.Port)
+
+	config := api.DefaultConfig()
+	config.Address = address
+	config.HttpClient = cleanhttp.DefaultClient()
+	config.HttpClient.Transport.(*http.Transport).TLSClientConfig = cores[0].TLSConfig
+	client, err := api.NewClient(config)
```

Comment:
Any reason not to just use the client from cores[0].Client rather than all of this boilerplate?